kartothek/io/dask/_update.py

```diff
@@ -38,17 +40,18 @@ def _update_dask_partitions_shuffle(
 
     final = []
     for lower, upper in zip(lower_bounds, upper_bounds):
-        chunk_ddf = ddf.partitions[lower : upper + 1]
+        lower_int = cast(int, lower)
```

Comment:
I think this might be technically wrong, since `upper` and `lower` are from a numpy array and might be numpy integers and not python integers. You can safely use `int(lower)` to actually enforce the type. `cast` only changes the mypy type w/o any actual runtime effect (also, it is unchecked and there is no checked cast in python 🤦‍♂ ).

Reply:
This is super weird. If I do `int(lower)` I get
```
No overload variant of "int" matches argument type "_T"
```

Reply:
🤦‍♂ python typing

Reply:
The int function is apparently defined as 

```
def int(self, x: Union[str, bytes, SupportsInt] = ...) -> int
```

Therefore, the Numpy types would need to use `SupportsInt`

still...